modules/apps/commerce/commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/AvailabilityLabelTag.java

```diff
@@ -15,22 +15,50 @@
 package com.liferay.commerce.frontend.taglib.servlet.taglib;
 
 import com.liferay.commerce.frontend.taglib.internal.servlet.ServletContextUtil;
+import com.liferay.commerce.product.constants.CPContentContributorConstants;
+import com.liferay.commerce.product.content.util.CPContentHelper;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
 import com.liferay.taglib.util.IncludeTag;
 
+import java.util.Map;
+
 import javax.servlet.http.HttpServletRequest;
+import javax.servlet.jsp.JspException;
 import javax.servlet.jsp.PageContext;
 
 /**
  * @author Gianmarco Brunialti Masera
  */
 public class AvailabilityLabelTag extends IncludeTag {
 
-	public int getStockQuantity() {
-		return _stockQuantity;
+	@Override
+	public int doStartTag() throws JspException {
+		try {
+			Map<String, String> availabilityMap =
+				_cpContentHelper.getAvailabilityMap(getRequest());
```

Comment:
@ivicac I know that is not related, please put in the backloag to delete the request fro the paraemter list, we need to make it reciving  needed parameters. we have to deprecate old ones